codalab/worker/dependency_manager.py

```diff
@@ -187,35 +251,46 @@ def loop(self):
         self._main_thread.start()
 
     def stop(self):
-        logger.info('Stopping local dependency manager')
+        logger.info('Stopping local dependency manager...')
         self._stop = True
         self._downloading.stop()
         self._main_thread.join()
-        logger.info('Stopped local dependency manager')
+        self._state_lock.release()
+        logger.info('Stopped local dependency manager.')
+
+    def _transition_dependencies(self):
+        with self._state_lock:
+            dependencies, paths = self._fetch_state(default={'dependencies': {}, 'paths': set()})
```

Comment:
Remove default and add a try-except to check for when the state was not properly loaded

```suggestion
            dependencies, paths = self._fetch_state()
```